drf_spectacular/openapi.py

```diff
@@ -1016,6 +1023,47 @@ def _get_response_for_code(self, serializer, status_code, media_types=None):
             # https://github.com/OAI/OpenAPI-Specification/blob/master/versions/3.0.3.md#responseObject
             'description': ''
         }
+        if headers:
+            response["headers"] = headers
+        return response
+
+    def _get_response_headers_for_code(self, status_code) -> dict:
+        """
+        Extract the response header parameters.
+
+        TODO: on HTTP 201 with generics.CreateView, DRF tries to set the Location header
+        param automatically. Include that here?
+        """
+        parameters = []
+
+        # select the relevant response header parameters
+        for parameter in self.get_override_parameters():
+            if not isinstance(parameter, OpenApiParameter):
+                continue
+            if not parameter.response:
+                continue
+
+            # check if the parameter is always present, or only for this status code
+            if parameter.response is True or status_code in parameter.response:
+                parameters.append(parameter)
```

Comment:
can we move this in the for loop below. then this method body is structurally identical to `_process_override_parameters`. 

one more thing i noticed is the additional stuff that can be added https://github.com/tfranzel/drf-spectacular/blob/97f2dc928572b495d5b58905ac874af9aa80614d/drf_spectacular/openapi.py#L142 not sure if everything applies here also. you may ignore this if you wish. i can replace that later if you lack the time.

Reply:
Yes, looking at the [spec](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/3.0.2.md#headerObject) it seems like we should call `build_parameter_type` after all, and then drop the `name` and `in` keys.

There are some rules about what's allowed for `style` etc., but that's up to the end-user to do the right thing, I think. At this point, they're using `extend_schema` already anyway.